src/proxy/core/routers/mock/mock.py

```diff
@@ -0,0 +1,148 @@
+import asyncio
+import json
+import os
+from typing import Annotated, Optional
+
+import jwt
+from fastapi import APIRouter, Depends, Header, HTTPException
+from fastapi.responses import StreamingResponse
+from fxa.errors import TrustError
+from fxa.oauth import Client
+
+from ...auth.fxa_auth import authorize_request
+from ...classes import AuthorizedChatRequest, ChatRequest
+from ...config import env
+from ...utils import get_or_create_user
+
+router = APIRouter()
+
+fxa_url = (
+	"https://api-accounts.stage.mozaws.net/v1"
+	if env.DEBUG
+	else "https://oauth.accounts.firefox.com/v1"
+)
+fxa_client = Client(env.CLIENT_ID, env.CLIENT_SECRET, fxa_url)
+
+
+def verify_jwt_token_only(x_fxa_authorization: Annotated[str | None, Header()]):
+	"""
+	Verify JWT token using pyfxa's _verify_jwt_token method without making POST calls.
+	This is useful for load testing where we want to verify token validity locally.
+	"""
+	if not x_fxa_authorization:
+		raise HTTPException(status_code=401, detail="Missing FxA authorization header")
+
+	try:
+		token = x_fxa_authorization.removeprefix("Bearer ").split()[0]
+	except Exception:
+		raise HTTPException(
+			status_code=401, detail="Invalid authorization header format"
+		)
+
+	try:
+		# Get JWKS keys from FxA server (this is a GET request, not POST)
+		keys = fxa_client.apiclient.get("/jwks").get("keys", [])
+
+		# Try to verify the token using each key
+		for key in keys:
+			try:
+				result = fxa_client._verify_jwt_token(json.dumps(key), token)
+				return result
+			except jwt.exceptions.InvalidSignatureError:
+				# Try next key if signature doesn't match
+				continue
+			except jwt.exceptions.PyJWTError as e:
+				# Other JWT errors (expired, malformed, etc.)
+				raise HTTPException(
+					status_code=401, detail=f"JWT verification failed: {str(e)}"
+				)
+
+		# If we get here, no key matched the signature
+		raise HTTPException(status_code=401, detail="Invalid token signature")
+
+	except TrustError as e:
+		raise HTTPException(status_code=401, detail=f"Token trust error: {str(e)}")
+	except Exception as e:
+		raise HTTPException(
+			status_code=401, detail=f"Token verification failed: {str(e)}"
+		)
+
+
+@router.post(
+	"/chat/completions", description="Mock LiteLLM endpoint with simulated latency."
+)
+async def chat_completion(
+	authorized_chat_request: Annotated[
+		Optional[AuthorizedChatRequest], Depends(authorize_request)
+	],
+):
+	user_id = authorized_chat_request.user
+	if not user_id:
+		raise HTTPException(
+			status_code=400,
+			detail={"error": "User not found from authorization response."},
+		)
+
+	user, _ = await get_or_create_user(user_id)
+	if user.get("blocked"):
+		raise HTTPException(status_code=403, detail={"error": "User is blocked."})
+
+	latency_ms = int(os.getenv("MOCK_LATENCY_MS", "200"))
```

Comment:
This should be added to the `config.py` and accessed with `env.MOCK_LATENCY_MS`

Reply:
Or if we want to refactor that to some other structure, I'd be fine with that too